src/app/modules/form/admin-form/admin-form.controller.ts

```diff
@@ -1640,6 +1642,97 @@ export const _handleCreateFormField: RequestHandler<
   )
 }
 
+/**
+ * NOTE: Exported for testing.
+ * Private handler for POST /forms/:formId/logic
+ * @precondition Must be preceded by request validation
+ * @security session
+ *
+ * @returns 200 with created logic object when successfully created
+ * @returns 403 when user does not have permissions to create logic
+ * @returns 404 when form cannot be found
+ * @returns 422 when user in session cannot be retrieved from the database
+ * @returns 500 when database error occurs
+ */
+export const _handleCreateLogic: RequestHandler<
+  { formId: string },
+  LogicDto | ErrorDto,
+  LogicDto
+> = (req, res) => {
+  const { formId } = req.params
+  const createdLogic = req.body
+  const sessionUserId = (req.session as Express.AuthedSession).user._id
+
+  // Step 1: Retrieve currently logged in user.
+  return (
+    UserService.getPopulatedUserById(sessionUserId)
+      .andThen((user) =>
+        // Step 2: Retrieve form with write permission check.
+        AuthService.getFormAfterPermissionChecks({
+          user,
+          formId,
+          level: PermissionLevel.Write,
+        }),
+      )
+      // Step 3: Create form logic
+      .andThen((retrievedForm) =>
+        AdminFormService.createFormLogic(retrievedForm, createdLogic),
+      )
+      .map((createdLogic) => res.status(StatusCodes.OK).json(createdLogic))
+      .mapErr((error) => {
+        logger.error({
+          message: 'Error occurred when creating form logic',
+          meta: {
+            action: 'handleCreateLogic',
+            ...createReqMeta(req),
+            userId: sessionUserId,
+            formId,
+            createdLogic,
+          },
+          error,
+        })
+        const { errorMessage, statusCode } = mapRouteError(error)
+        return res.status(statusCode).json({ message: errorMessage })
+      })
+  )
+}
+
+/**
+ * Handler for POST /forms/:formId/logic
+ */
+export const handleCreateLogic = [
+  celebrate({
+    [Segments.BODY]: Joi.object({
+      logicType: Joi.string()
+        .valid(...Object.values(LogicType))
+        .required(),
+      conditions: Joi.array()
+        .items(
+          Joi.object({
+            field: Joi.string().required(),
+            state: Joi.string()
+              .valid(...Object.values(LogicConditionState))
+              .required(),
+            value: Joi.any().required(),
+            ifValueType: Joi.valid(...Object.values(LogicIfValue)),
+          }).unknown(true),
+        )
+        .required(),
+      show: Joi.alternatives().conditional('logicType', {
+        is: LogicType.ShowFields,
+        then: Joi.array().items(Joi.string()).required(),
+      }),
+      preventSubmitMessage: Joi.alternatives().conditional('logicType', {
+        is: LogicType.PreventSubmit,
+        then: Joi.string(),
+      }),
+      // Allow other field related key-values to be provided and let the model
+      // layer handle the validation.
+    }).unknown(true),
```

Comment:
same as above wrt `unknown`

Reply:
I left `unknown` so that we don't block requests that have more keys than we expected, which are harmless (the validation is there to ensure that the keys which we expect are present in the expected shape)

Reply:
ah ok, thanks for pointing this out! just as an aside, we're not expecting any more keys beyond `_id?` though, right?

Reply:
nope, not expecting any. for create form logic there should also not be _id in the request